csharp/src/Microsoft.ML.OnnxRuntime/NamedOnnxValue.cs

```diff
@@ -16,19 +16,22 @@ public class NamedOnnxValue
     {
         protected Object _value;
         protected string _name;
+        protected IntPtr _onnxValue;
+        protected MemoryHandle _pinnedMemoryHandle;
 
         protected NamedOnnxValue(string name, Object value)
         {
             _name = name;
             _value = value;
+            _onnxValue = IntPtr.Zero;
         }
 
         public static NamedOnnxValue CreateFromTensor<T>(string name, Tensor<T> value)
         {
             return new NamedOnnxValue(name, value);
         }
 
-        public string Name { get { return _name; } }
+        public string Name { get { return _name; } set { _name = value; } }
```

Comment:
Support a setter

Reply:
This adds a new public API -- but looks like it's necessary if we want to pass outputs of a model (renamed) as inputs back into the model.